### PR TITLE
Add structured logging for KMC

### PR DESCRIPTION
**Pull request by thandayuthapani**
**Description**
Add structured log for Kyma Metrics Collector

Changes proposed in this pull request:

- Introduce new structured logging scheme for KMC

**Related issue(s)**
Fixes https://github.com/kyma-project/control-plane/issues/847

### CODE REVIEW

File: components/kyma-metrics-collector/pkg/logger/logger.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+// LoggerOpt represents a function that returns a log pair instance when executed.
+type LoggerOpt func() pair
+
+// WithAction returns a LoggerOpt for the given action.
+func WithAction(action string) LoggerOpt {
+	return func() pair {
+		return pair{keyAction, action}
+	}
+}
```
Comment (marcobebway) [inline components/kyma-metrics-collector/pkg/logger/logger.go:93]: Do we need these?

Comment (thandayuthapani) [reply components/kyma-metrics-collector/pkg/logger/logger.go:93]: Just using the same scheme as evening reconciler.  If there is no necessity for this in KMC that will be removed.

File: components/kyma-metrics-collector/pkg/edp/client_test.go [hunk 2]
```diff
@@ -55,7 +56,7 @@ func TestClient(t *testing.T) {
 	edpURL, err := url.ParseRequestURI(srv.URL)
 	g.Expect(err).Should(gomega.BeNil())
 
-	edpClient := NewClient(config, logrus.New())
+	edpClient := NewClient(config, logger.NewLogger(zapcore.DebugLevel))
```
Comment (k15r) [inline components/kyma-metrics-collector/pkg/edp/client_test.go:59]: is there a reason to always use debug?

File: components/kyma-metrics-collector/pkg/edp/client_test.go [hunk 3]
```diff
@@ -98,7 +99,7 @@ func TestClientRetry(t *testing.T) {
...
```
Comment (k15r) [inline components/kyma-metrics-collector/pkg/edp/client_test.go:102]: please dont default to debug

File: components/kyma-metrics-collector/pkg/edp/client_test.go (hunk 2 above)
Comment (thandayuthapani) [reply components/kyma-metrics-collector/pkg/edp/client_test.go]: Fixed

File: components/kyma-metrics-collector/pkg/edp/client_test.go (hunk 3 above)
Comment (thandayuthapani) [reply components/kyma-metrics-collector/pkg/edp/client_test.go]: Fixed

**Review by k15r — APPROVED** [review]

**Pull request merged**